code/modules/research/research.dm

```diff
@@ -44,11 +43,13 @@ research holder datum.
 **	Includes all the helper procs and basic tech processing.  **
 ***************************************************************/
 
+// Global design lists
+var/global/list/designs = null
+var/global/list/designs_protolathe_categories = list()
+var/global/list/designs_imprinter_categories = list()
```

Comment:
I'd prefer it if we could find a way to do this without keeping global lists of all the designs.

If we can move this whole thing into a unit test, and these lists *are* necessary, then we can just keep them around for the duration of the unit test.

Reply:
The point of making it a global list is so that it doesn't need to be initialized more than once which should speed up load times (as every single R&D server, console, etc. would be re-initializing all the designs). These lists or their related objects are never changed, and yet they hold hundreds of objects, I think it's better if they aren't duplicate.

Reply:
Ok. I need to look at it a bit deeper then, thanks.

Reply:
Though TBF I did originally do it to do the check just once. At the moment it's not necessary for this PR to go through. But I've tested it and everything still seems to work as expected, so whatever.

Reply:
I took a better look at how the research designs thing works. I just wanted to make sure designs weren't being new'd multiple times and kept in different places. And it all looks good 👍

Reply:
There's still definitely a lot of room for improvement, but I don't think this is the right place for this.

I also noticed some talks of complete research overhaul, though I'm not sure if anyone is actually working on anything.